tc/core/polyhedral/schedule_tree.h

```diff
@@ -126,6 +126,7 @@ namespace detail {
 
 std::ostream& operator<<(std::ostream& os, const ScheduleTree& tree);
 
+template <typename T>
```

Comment:
while you're modifying this function, shouldn't we make it return the flattened tree for API consistency

Reply:
This change is irrelevant to the goal of this PR.  Lately, we have been enforcing the policy of not having irrelevant commits in the PR.  I'm okay with either way, but let's be consistent.

Reply:
sure let's keep PR consistency first